### PR TITLE
fix: delegated signatures: check every field of txs and roundtrip eth <-> FIL

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

https://github.com/filecoin-project/ref-fvm/issues/1433

## Proposed Changes

- Make EthTxArgsFromMessage much stricter
- Roundtrip Eth <-> FIL messages during siggy validation

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/types/ethtypes/eth_transactions.go [hunk 1]
```diff
@@ -103,13 +121,79 @@ func NewEthTxArgsFromMessage(msg *types.Message) (EthTxArgs, error) {
...
+		Value:      args.Value,
+		GasLimit:   int64(args.GasLimit),
+		GasFeeCap:  args.MaxFeePerGas,
+		GasPremium: args.MaxPriorityFeePerGas,
+		Method:     method,
+		Params:     params,
+	}, nil
+}
+
 func (tx *EthTxArgs) ToSignedMessage() (*types.SignedMessage, error) {
```
Comment (arajasek) [inline chain/types/ethtypes/eth_transactions.go:197]: Oops, I didn't notice the method right after this one does the same thing. We'll need to either consolidate these methods or drop the new one before merging.

**Comment by arajasek** [comment]
I'd like to write tests covering a lot of the edge-cases (eg. bad Version, bad params), but I'm not sure whether that should block landing this PR.

File: chain/types/ethtypes/eth_transactions.go [hunk 2]
```diff
@@ -89,12 +91,26 @@ func NewEthTxArgsFromMessage(msg *types.Message) (EthTxArgs, error) {
...
+						msg.Method,
+						builtintypes.MethodsEVM.InvokeContract)
+			}
+
+			params, err = cbg.ReadByteArray(paramsReader, uint64(len(msg.Params)))
 			if err != nil {
-				return EthTxArgs{}, err
+				return EthTxArgs{}, xerrors.Errorf("failed to read params byte array: %w", err)
+			}
+
```
Comment (Stebalien) [inline chain/types/ethtypes/eth_transactions.go:110]: We need to assert that `params` is non-empty here. Otherwise, this _should_ have been a method send and someone pulled a fast one.

**Review by Stebalien — APPROVED** [review]
